src/runtime/client-event-manager.js

```diff
@@ -29,6 +29,28 @@ function createEventErrorMessage(valueName, value) {
   return 'Event has an invalid ' + valueName + '(' + value + ')';
 }
 
+/**
+ * Utility function to create a ClientEvent proto.
+ * @param {!AnalyticsEvent} eventType
+ * @param {?boolean=} isFromUserAction
+ * @param {!EventOriginator=} eventOriginator
+ * @param {?Object=} additionalParameters
+ * @returns {!../api/client-event-manager-api.ClientEvent}
+ */
+export function createClientEvent(
```

Comment:
Good point. In talking with Matt, we came up with the plan of adding a "logSwgEvent" method to the EventManager that accepts AnalyticsEvent, isFromUserAction, and additionParams. That way, we can just call eventManager().logSwgEvent directly instead of having this wrapper function, and since EventManager is only being used by the SWG client, we don't have to expose EventOriginator.

Reply:
Well it shouldn't be necessary to add a new API to default one parameter. Again logEvent() should be able to handle it. Unless I"m missing something.

Reply:
I'm a bit confused. Are you proposing to alter logEvent's arguments such that it accepts (AnalyticsEvent, isFromUserAction, EventOriginator, and additionalParams) instead of accepting a ClientEvent object?

Reply:
that's one way. 
// logEvent(event, active, originator, jsonParams)
OR
// If ClientEvent must be fully formed before sending to EM, then define default params instead of a helper function
DEFAULT_ACTIVE = false
DEFAULT_ORIGINATOR = SWG_CLIENT
DEFAULT_PARAMS = null
so the callers can construct a default event using these instead of invoking a helper function
logEvent({AnalyticsEvent.IMPRESSION_PAYWALL, ClientEventManager.DEFAULT_ACTIVE, ClientEventManager.DEFAULT_ORIGINATOR, ClientEventManager.DEFAULT_PARAMS})

OR
// allow logEvent to just accept AnalyticsEvent as an optional argument
/**
 * Either event or optEvent should be provided by the caller
 * @param {?ClientEvent} event
 * @param {AnalyticsEvent=} optEvent
 */
logEvent(clientEvent, optEvent) {
  let event = clientEvent;
  if (!event) {
    event = {optEvent, DEFAULT_ACTIVE, DEFAULT_ORIGINATOR, DEFAULT_PARAMS}
  }
  // do other stuff.
}
The caller would then do
logEvent(null, AnalyticsEvent.IMPRESSION_PAYWALL);

OR
// make ClientEvent a class with no helper functions but public fields
export class ClientEvent {
  /**
   * @param {!ClientEventType} type
   */
  constructor(type) {
    /** {!ClientEventType} */
    this.eventType = null;
    /** {boolean} */
    this.isFromUserAction = false;
    /** {EventOriginator} */
    this.eventOriginator = SWG_CLIENT;
    /** {?JsonObject} */
    this.additionalParams = null;
  }
}

So callers may do:
logEvent(new ClientEvent(AnalyticsEvent.IMPRESSION_PAYWALL));
OR do this instead:
 const event = new ClientEvent(AnalyticsEvent.IMPRESSION_PAYWALL);
  event.isFromUserAction = true;
  event.eventOriginator = PUBLISHER_CLIENT
  event.additionalParams = myJsonParams;
  logEvent(event);

All of the above are some API change or other. But logEvent() API IMHO is the one that needs fixing, not adding new helpers or APIs to solve the problem.

Reply:
My issue with changing the logEvent method or the ClientEvent object is that AMP is already using this API as is. Changing the public API would make it so that we would have to change both the SwG code and the AMP code concurrently, which I believe to be out of the scope of this PR. Although this could be done, I think it would be overkill to do so because this change would make the API only slightly more convenient and I would guess that the only users of this API will be in SwG anyways as EventManager will be getting wrapped by LoggerApi for publisher use.

The reason I like the logSwgEvent method is because it finds a middle ground between not changing a public API and making it easier to use inside of SwG. By adding this method to the implementation and not the API, we can use it inside of Runtime--which is where EventManager will mostly be used--and hide it from the public. This makes it so I can just use one function call (logSwgEvent(...) instead of two (logEvent(createClientEvent(...), and we won't have to change any AMP code.

What do you think?

Reply:
I agree that logSwGEvent() or createClientEvent() is a good solution without having to change API. Because internally we can get away with a helper or alternate API like logSwgEvent. But the need for this internally indicates that maybe even AMP code is finding this API difficult to use? If that's the case, then we are better off fixing this sooner than later, before too many callers are calling it. Agree that whatever we decide is out of scope of this PR we just need to document our decision and then follow up with another change if needed. I really feel like decision should be based on ease of use in AMP. I have approved this PR so that you can document your final decision and merge it.

Reply:
Thank you for your approval. I will add the logSwgEvent and plan on leaving the logEvent/EventManager API as is as I don't think it is worth the effort given that we are behind on the logging project right now.